test/jdk/java/nio/charset/Charset/IllegalCharsetName.java

```diff
@@ -22,46 +22,86 @@
  */
 
 /* @test
- * @bug 6330020 8184665
+ * @bug 4448594 4786884 6330020 8184665 8260265
  * @summary Ensure Charset.forName/isSupport throws the correct exception
  *          if the charset names passed in are illegal.
+ * @run junit IllegalCharsetName
  */
 
-import java.nio.charset.*;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.IllegalCharsetNameException;
+import java.nio.charset.UnsupportedCharsetException;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class IllegalCharsetName {
-    public static void main(String[] args) throws Exception {
-        String[] illegalNames = {
-            ".",
-            "_",
-            ":",
-            "-",
-            ".name",
-            "_name",
-            ":name",
-            "-name",
-            "name*name",
-            "name?name"
-        };
-        for (int i = 0; i < illegalNames.length; i++) {
-            try {
-                Charset.forName(illegalNames[i]);
-                throw new Exception("Charset.forName(): No exception thrown");
-            } catch (IllegalCharsetNameException x) { //expected
-            }
 
-            try {
-                Charset.isSupported(illegalNames[i]);
-                throw new Exception("Charset.isSupported(): No exception thrown");
-            } catch (IllegalCharsetNameException x) { //expected
-            }
-        }
+    // Charset.forName should throw an exception when passed "default"
+    @Test
+    public void defaultCharsetTest() {
+        assertThrows(UnsupportedCharsetException.class,
+                () -> Charset.forName("default"));
+    }
```

Comment:
Seems incorrect to merge this test into `IllegalCharsetName.java` because it is throwing `UnsupportedCharsetException`.

Reply:
I moved it back to RegisteredCharsets.java